src/FileStorage.php

```diff
@@ -58,18 +58,8 @@ final class FileStorage implements Storage, TransferableStorage
      * @param array                      $dir
      * @param array                      $options
      */
-    public function __construct($writer, $reader, array $dir, array $options = [])
+    public function __construct(TranslationWriterInterface $writer, TranslationReaderInterface $reader, array $dir, array $options = [])
```

Comment:
Removing the possibility to use `TranslationLoader` here is a BC break Im afraid.